src/printer.js

```diff
@@ -2384,12 +2384,25 @@ function printBinaryishExpressions(path, parts, print) {
     ) {
       // Flatten them out by recursively calling this function. The
       // printed values will all be appended to `parts`.
-      path.call(left => printBinaryishExpressions(left, parts, print), "left");
+      path.call(left => printBinaryishExpressions(left, parts, print, options, /* isNested */ true), "left");
     } else {
       parts.push(path.call(print, "left"));
     }
 
     parts.push(" ", node.operator, line, path.call(print, "right"));
+
+    // The root comments are already printed, but we need to manually print
+    // the other ones since we don't call the normal print on BinaryExpression,
+    // only for the left and right parts
+    if (isNested && node.comments) {
+      parts.push(
+        comments.printComments(
```

Comment:
> p => "" is what prints the node at path. So in this case it's not going to print anything ;)

Oh, right, hah. You can do this now because I inlined the comment printing logic (it used to go through the same big switch case. Looking at this more now!